xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift

```diff
@@ -441,15 +441,12 @@ private struct EditProfileSheetModifier: ViewModifier {
                 store.actions.compactMap(AppAction.from),
                 perform: app.send
             )
-            .onReceive(app.actions, perform: { action in
-                switch (action) {
-                case .succeedIndexPeer(let peer) where peer.identity == store.state.user?.did:
-                    store.send(.refresh(forceSync: false))
-                    break
-                case _:
-                    break
-                }
-            })
+            .onReceive(
+                app.actions.compactMap { action in
+                    UserProfileDetailAction.fromAppAction(action: action)
```

Comment:
Nit: you could pass this function directly to compactMap instead of wrapping in closure.